src/views/dataset/MultiSourceConfiguration.vue

```diff
@@ -1352,17 +1473,44 @@ export default class MultiSourceConfiguration extends Vue {
         transcode: this.transcode,
         eventCallback,
       });
-      // Schedule caches after adding multisource (and transcoding)
-      this.store.scheduleTileFramesComputation(datasetId);
-      this.store.scheduleMaxMergeCache(datasetId);
-      this.store.scheduleHistogramCache(datasetId);
 
       if (!itemId) {
         throw new Error("Failed to add multi source");
       }
+
+      // Store dimension labels as separate metadata on the dataset folder
+      try {
+        const dimensionLabels = {
+          xy: xyLabels,
+          z: zLabels,
+          t: tLabels,
+        };
+        await this.store.api.updateDatasetMetadata(datasetId, {
+          dimensionLabels: dimensionLabels,
+        });
+        // Invalidate the folder cache so the updated metadata is fetched
+        await this.store.girderResources.ressourceChanged(datasetId);
+      } catch (labelError) {
+        logError("Failed to store dimension labels (non-fatal):", labelError);
+        // Don't fail the whole process if label storage fails
+      }
+
+      // Schedule caches after successful metadata upload
+      this.store.scheduleTileFramesComputation(datasetId);
+      this.store.scheduleMaxMergeCache(datasetId);
+
+      // Add a small delay before scheduling histogram cache to ensure the item is processed
+      setTimeout(async () => {
+        try {
+          await this.store.scheduleHistogramCache(datasetId);
+        } catch (error) {
+          logError("Failed to schedule histogram cache:", error);
+        }
+      }, 2000);
```

Comment:
That will not play well with the cloud deployment, you cannot expect a time limit to always work  
If we always schedule the histogram cache after the item is processed, maybe the backend could be responsible for scheduling it when the item is done processing ?